examples/decoding/plot_decoding_spatio_temporal_source.py

```diff
@@ -58,93 +64,51 @@
-clf = SVC(C=1, kernel='linear')
-cv = ShuffleSplit(len(X), n_splits, test_size=0.2)
-
-# setup feature selection and classification pipeline
-from sklearn.feature_selection import SelectKBest, f_classif  # noqa
-from sklearn.pipeline import Pipeline  # noqa
-
-# we will use an ANOVA f-test to preselect relevant spatio-temporal units
-feature_selection = SelectKBest(f_classif, k=500)  # take the best 500
-# to make life easier we will create a pipeline object
-anova_svc = Pipeline([('anova', feature_selection), ('svc', clf)])
-
-# initialize score and feature weights result arrays
-scores = np.zeros(n_splits)
-feature_weights = np.zeros([n_vertices, n_times])
-
-# hold on, this may take a moment
-for ii, (train, test) in enumerate(cv):
-    anova_svc.fit(X[train], y[train])
-    y_pred = anova_svc.predict(X[test])
-    y_test = y[test]
-    scores[ii] = np.sum(y_pred == y_test) / float(len(y_test))
-    feature_weights += feature_selection.inverse_transform(clf.coef_) \
-        .reshape(n_vertices, n_times)
-
-print('Average prediction accuracy: %0.3f | standard deviation:  %0.3f'
-      % (scores.mean(), scores.std()))
-
-# prepare feature weights for visualization
-feature_weights /= (ii + 1)  # create average weights
-# create mask to avoid division error
-feature_weights = np.ma.masked_array(feature_weights, feature_weights == 0)
-# normalize scores for visualization purposes
-feature_weights /= feature_weights.std(axis=1)[:, None]
-feature_weights -= feature_weights.mean(axis=1)[:, None]
-
-# unmask, take absolute values, emulate f-value scale
-feature_weights = np.abs(feature_weights.data) * 10
+# Decoding in sensor space using a logistic regression
+
+# Retrieve source space data into an array
+X = np.array([stc.lh_data for stc in stcs])  # only keep left hemisphere
+y = epochs.events[:, 2]
+
+# prepare a series of classifier applied at each time sample
+clf = make_pipeline(StandardScaler(),  # z-score normalization
+                    SelectKBest(f_classif, k=500),  # select features for speed
+                    LinearModel(LogisticRegression(C=1)))
+time_decod = SlidingEstimator(clf, scoring='roc_auc')
+
+# Run cross-validated decoding analyses:
+scores = cross_val_multiscore(time_decod, X, y, cv=7, n_jobs=1)
+
+# Plot average decoding scores of 10 splits
+fig, ax = plt.subplots(1)
+ax.plot(epochs.times, scores.mean(0), label='score')
+ax.axhline(.5, color='k', linestyle='--', label='chance')
+ax.axvline(0, color='k')
+plt.legend()
+
+###############################################################################
+# To investigate weights, we need to retrieve the patterns of a fitted model
+
+# The fitting needs not be cross validated because the weights are based on
+# the training sets
+time_decod.fit(X, y)
+
+# Retrieve patterns after inversing the z-score normalization step:
+patterns = get_coef(time_decod, 'patterns_', inverse_transform=True)
 
 vertices = [stc.lh_vertno, np.array([], int)]  # empty array for right hemi
```

Comment:
This is by the way crashing on python 3.
One should now explicitly define the vertex numbers or get it from stcs[0] since it is now a list.